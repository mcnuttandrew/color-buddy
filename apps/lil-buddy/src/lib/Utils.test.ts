import { expect, test } from "vitest";
import {
  modifyLint,
  MODIFY_LINT_TARGET_KEY,
  MODIFY_LINT_DELETE,
} from "./utils";

const exampleLintStr = `{
    "all": {
        "predicate": {
            "==": { "right": true, "left": {"inGamut": "a"} }
        }
    }
}
`;
test("modifyLint", () => {
  expect(modifyLint([], "asd", exampleLintStr)).toMatchSnapshot();
  expect(modifyLint(["1"], "asd", exampleLintStr)).toMatchSnapshot();
  expect(modifyLint(["all"], "asd", exampleLintStr)).toMatchSnapshot();
  expect(
    modifyLint(["all", "predicate"], "asd", exampleLintStr)
  ).toMatchSnapshot();
  expect(
    modifyLint(["all", "predicate", "=="], "asd", exampleLintStr)
  ).toMatchSnapshot();
  expect(
    modifyLint(["all", "predicate", "==", "left"], "asd", exampleLintStr)
  ).toMatchSnapshot();
});

test("modifyLint.non obj swap key case", () => {
  expect(
    modifyLint(
      ["all", "predicate", "==", "left", "inGamut", MODIFY_LINT_TARGET_KEY],
      "asd",
      exampleLintStr
    )
  ).toMatchSnapshot();
});

test("modifyLint delete", () => {
  expect(
    modifyLint(
      ["all", "predicate", "==", "left", "inGamut"],
      MODIFY_LINT_DELETE,
      exampleLintStr
    )
  ).toMatchSnapshot();
});
