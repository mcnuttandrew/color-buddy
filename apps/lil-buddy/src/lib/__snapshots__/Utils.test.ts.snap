// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`modifyLint 1`] = `"{"all":{"predicate":{"==":{"right":true,"left":{"inGamut":"a"}}}},"undefined":"asd"}"`;

exports[`modifyLint 2`] = `"{"1":"asd","all":{"predicate":{"==":{"right":true,"left":{"inGamut":"a"}}}}}"`;

exports[`modifyLint 3`] = `"{"asd":{"predicate":{"==":{"right":true,"left":{"inGamut":"a"}}}}}"`;

exports[`modifyLint 4`] = `"{"all":{"asd":{"==":{"right":true,"left":{"inGamut":"a"}}}}}"`;

exports[`modifyLint 5`] = `"{"all":{"predicate":{"asd":{"right":true,"left":{"inGamut":"a"}}}}}"`;

exports[`modifyLint 6`] = `"{"all":{"predicate":{"==":{"right":true,"asd":{"inGamut":"a"}}}}}"`;

exports[`modifyLint delete 1`] = `"{"all":{"predicate":{"==":{"right":true}}}}"`;

exports[`modifyLint.non obj swap key case 1`] = `"{"all":{"predicate":{"==":{"right":true,"left":{"asd":"a"}}}}}"`;
